tsconfig.json

```diff
@@ -6,6 +6,8 @@
     "allowJs": true,
     "checkJs": true,
     "strict": true,
+    // TODO: turn this off to be fully `strict`.
+    "useUnknownInCatchVariables": false,
```

Comment:
this would [make all catch clause variables `unknown`](https://devblogs.microsoft.com/typescript/announcing-typescript-4-4-rc/#use-unknown-catch-variables), so we would always have to do `typeof`/`instanceof`/`x in err` checks before taking properties off of errors.

That's the right way to do it in JS, where code can throw whatever it wants, but we probably aren't ready to commit to that yet

Reply:
"right way" when you have 3p js that can do whatever it wants... I wish there were a way at the language level to say "this code doesn't do terrible error handling". Like, `use strict` should have disallowed throwing random crap :)